core/sequencer/election_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sequencer
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/keytransparency/core/domain"
+	"github.com/google/keytransparency/core/mutator"
+	"github.com/google/trillian/monitoring"
+	"github.com/google/trillian/util/election"
+)
+
+func TestSingleMastership(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	createMetrics(monitoring.InertMetricFactory{})
+	d := &domain.Domain{DomainID: "testdomain"}
+	c := NewController(ctx, d, &election.NoopFactory{}, 1*time.Hour, monitoring.InertMetricFactory{})
+
+	// * Unblock and relinquish mastership.
+
+	// 1. Run election.
+	if err := c.RunWhileMaster(ctx,
+		// 2. Create new receiver.
+		func(ctx context.Context, d *domain.Domain, checkMastership func() error) (mutator.Receiver, error) {
+			go func() {
+				// 3. Receiver receives callback.
+				if err := checkMastership(); err != nil {
```

Comment:
I'm struggling to understand why this invocation should work. I don't see where you provider a non-nil `checkMastership` to the `func`.